enhancements/sig-architecture/105-aws-iam-registration/README.md

```diff
@@ -0,0 +1,464 @@
+
+```
+% clusteradm accept \
+     --registration-auth=aws-irsa \
+     --clusters managed-0
+```
+
+
+#### Story 5 - Managed cluster administrator un-joins a cluster from the hub using AWS IRSA authentication strategy
+
+It must be possible for the cluster administrator to specify they wish to use `aws-irsa` authentication strategy in the `clusteradm unjoin` command to un-register with hub. The default authentication strategy will be `csr`:
+
+```
+% clusteradm unjoin \
+     --registration-auth=aws-irsa \
+     --cluster-name managed-0
+```
+
+#### Story 6 - EKS Hub administrator must accept the registration request using aws-irsa authentication strategy only, using csr authentication strategy will throw an error
+
+Since EKS doesn't support CSR based client auth and, self managed or clusters running on other cloud providers will not support AWS IRSA, one hub cluster will not support both the authentication strategies at the same time. Running `clusteradm accept` command on EKS with --registration-auth=csr will throw an error as there is no csr to approve, provided both `clusteradm init` and `clusteradm join` were run using --registration-auth=aws-irsa on hub and managed cluster respectively. 
+
+
+#### Story 7 - Procedure for managed cluster administrator to follow to refresh the token
+
+The temporary IAM credentials, for the role assumed by pod, obtained by klusterlet-agent from sts will have TTL of 1 hour by default, which will be [renewed](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Credentials.html#:~:text=Occasionally%20credentials%20can%20expire%20in%20the%20middle%20of%20a%20long%2Drunning%20application.%20In%20this%20case%2C%20the%20SDK%20will%20automatically%20attempt%20to%20refresh%20the%20credentials%20from%20the%20storage%20location%20if%20the%20Credentials%20class%20implements%20the%20refresh()%20method.) by AWS SDK automatically. More info can be found [here](https://docs.aws.amazon.com/eks/latest/userguide/pod-configuration.html).
+
+The EKS kubeconfig will use these temporary IAM creds to generate temporary sts token using `aws eks get-token` command present in kubeconfig with [TTL of 15 mins](https://aws.github.io/aws-eks-best-practices/security/docs/iam/#:~:text=The%20token%20has%20a%20time,each%20time%20the%20token%20expires.). This sts token authenticates every request sent to EKS api-server, and is automatically renewed by kubeconfig on expiry. Snippet of user from an EKS kubeconfig:
+```yaml
+users:
+- name: test-user
+  user:
+    exec:
+      apiVersion: client.authentication.k8s.io/v1beta1
+      command: aws
+      args:
+      - --region
+      - my-aws-region
+      - eks
+      - get-token
+      - --cluster-name
+      - my-cluster
```

Comment:
is this the cluster-name of the hub or the cluster-name of the spoke?

Reply:
it should be hub, will also update the proposal